File: playground/stations/4-lua-b64/job.lua
-- 三格：同步問 LLM（不經 kernel）→ 塞一段 binary 進 state → 寫結果檔。
local json = require "json"
local ENDPOINT = "__PLAY__/stations/1-ask-once/endpoint.json"

local function ask(state)
  local req = {messages = {{role = "user", content = "用三個字形容今天。"}}}
  local r = aos.llm(ENDPOINT, req, here .. "/answer.json")
  state.answer = aos.llm_text(r)
end

local function stash(state)
  state.blob = aos.b64("\255\254\0 not utf-8")   -- 不是合法 UTF-8 的位元組，存檔會長成 {"$b64":"..."}，讀回來還原
end

local function write(state)
  local f = assert(io.open(here .. "/result.json", "w"))
  f:write(json.encode(state), "\n"); f:close()
end

return {
  {name = "ask", fn = ask},
  {name = "stash", fn = stash},
  {name = "write", fn = write},
}

File: playground/stations/3-py-llm/job.py
# 三格：丟問題給 kernel 排隊 → 等結果檔 → 把答案存下來。每叫一次 aos-step-py 只跑一格。
import json, os

K = "__K__"
QUESTION = "用一句話介紹你自己。"

def ask(state):
    req = {"messages": [{"role": "user", "content": QUESTION}]}
    state["result"] = aos.llm_submit(K, req, "play-py-1")   # 同名同內容再丟＝拿回同一張，不會卡
    return aos.wait_for(state["result"])                     # 這格做完後開始等結果檔

def read(state):
    state["answer"] = json.load(open(state["result"]))["text"]

def save(state):
    open(os.path.join(here, "answer.txt"), "w").write(state["answer"] + "\n")
